data/sea/52-psv.h

```diff
@@ -107,7 +123,72 @@ static int INLINE psv_compare (const char *xs, size_t xs_size, const char *ys, s
     return xs_size - ys_size;
 }
 
-static ierror_loc_t psv_read_buffer (psv_state_t *s)
+/* Seek to a line where the entity changes, or EOF */
+static int INLINE psv_skip_to_next_entity (psv_state_t *s)
+{
+    const char  *end_ptr    = s->input_ptr + s->input_size;
+    const char  *line_ptr   = s->input_ptr;
+    const char  *n_ptr      = 0;
+
+    char   *entity_cur      = s->entity_cur;
+    size_t  entity_cur_size = s->entity_cur_size;
+
+    iint_t  fact_count      = s->fact_count;
+
+    for (;;) {
+        const size_t bytes_remaining = end_ptr - line_ptr;
+
+        n_ptr = memchr (line_ptr, '\n', bytes_remaining);
+
+        if (n_ptr == 0) {
+            s->input_remaining = bytes_remaining;
```

Comment:
ahh, I think I understand now. this will return, and psv_snapshot will loop and read more of the buffer, so if the next one has the same entity it will call this again. cool!